File: exercise/ts/week06-legacy-code/exercise3/src/orderProcessor.ts
import {Order} from "./order";

export class OrderProcessor {
    processOrders(orders: Order[]): void {
        for (const item of orders) {
            if (item.getStatus() === Order.UNPROCESSED) {
                if (item.getItems() > 5) {
                    // apply bulk discount
                    item.setTotal(item.getTotal() * 0.9);
                }
                item.setStatus(Order.PROCESSED);
            }
        }
    }
}

File: exercise/java/week06-legacy-code/exercise3/src/main/java/order/OrderProcessor.java
package order;

import java.util.List;

public class OrderProcessor {
    public void processOrders(List<Order> orders) {
        for (Order item : orders) {
            if (item.getStatus() == Order.UNPROCESSED) {
                if (item.getItems() > 5) {
                    // apply bulk discount
                    item.setTotal(item.getTotal() * 0.9);
                }
                item.setStatus(Order.PROCESSED);
            }
        }
    }
}